routing/matcher.go

```diff
@@ -135,6 +136,15 @@ func canonicalizeHeaderRegexps(hrx map[string][]*regexp.Regexp) map[string][]*re
 	return chrx
 }
 
+func extractWeight(r *Route) int {
+	//for _, p := range r.Predicates {
+	//	if w, ok := p.(weight.Predicate); ok {
+	// 		THIS IS IMPOSSIBLE >> import cycle not allowed
```

Comment:
indeed. Let me think of an approach...

Reply:
what i am thinking about is that this not so much the functionality of the predicate, but the routing package itself. We can see that the Match() method is always true. So we can either:

1. move the predicate to the routing package
2. use only the eskip name to identify the predicate while processing the route configuration and parse the arguments in the routing package

no better idea right now. @szuecs , what do you think?

Reply:
thought of it a bit, and I like now 2. better.

Reply:
I am with you, do 2.

Reply:
for example, it could look like this:

https://github.com/ruudk/skipper/pull/1